src/report.rs

```diff
@@ -351,6 +361,112 @@ impl CrashReport {
+            let start: usize = if debug.line > 5 {
+                debug.line as usize - 5
+            } else {
+                0
+            };
+            let mut lines: Vec<String> = file
+                .lines()
+                .skip(start)
+                .enumerate()
+                .take_while(|(i, _)| *i < 10)
+                .map(|(i, l)| {
+                    if let Ok(l) = l {
+                        format!("    {:<6} {}", start + i + 1, l.trim_end())
+                    } else {
+                        format!("    {:<6} Corrupted line", start + i + 1)
+                    }
+                })
+                .collect::<Vec<String>>();
+            let crash_line = debug.line as usize - start - 1;
+            if crash_line < lines.len() {
+                lines[crash_line].replace_range(..4, "--->");
+                return Some(lines);
+            }
+        }
+
+        None
+    }
+
+    /// Remove trusted frames from stack trace and reutrn it as `Stacktrace` struct
+    pub fn filtered_stacktrace(&self) -> Result<Stacktrace> {
+        let mut rawtrace: Stacktrace = Default::default();
+        if !self.asan_report.is_empty() {
+            rawtrace = if let Ok(parsed_trace) = AsanStacktrace::parse_stacktrace(&self.stacktrace)
+            {
+                parsed_trace
+            } else {
+                return Err(Error::Casr(
+                    "Error while parsing asan stacktrace".to_string(),
+                ));
+            };
+        }
+
+        if !self.python_report.is_empty() {
+            rawtrace =
+                if let Ok(parsed_trace) = PythonStacktrace::parse_stacktrace(&self.stacktrace) {
+                    parsed_trace
+                } else {
+                    return Err(Error::Casr(
+                        "Error while parsing python stacktrace".to_string(),
+                    ));
+                };
+        }
+
+        if rawtrace.is_empty() {
+            rawtrace = if let Ok(parsed_trace) = GdbStacktrace::parse_stacktrace(&self.stacktrace) {
+                parsed_trace
+            } else {
+                return Err(Error::Casr(
+                    "Error while parsing gdb stacktrace".to_string(),
+                ));
+            };
+        };
+
+        // For libfuzzer: delete functions below LLVMFuzzerTestOneInput
+        if let Some(pos) = &rawtrace
+            .iter()
+            .position(|x| x.function.contains("LLVMFuzzerTestOneInput"))
+        {
+            rawtrace.drain(pos + 1..);
+        }
+
+        if rawtrace.is_empty() {
+            return Err(Error::Casr(
+                "Current stack trace length is null".to_string(),
+            ));
+        }
+
+        // Get Proc mappings from Casr report
+        if !self.proc_maps.is_empty() {
+            let mappings = MappedFiles::from_gdb(self.proc_maps.join("\n"))?;
+            rawtrace.compute_module_offsets(&mappings);
+        }
+
+        rawtrace.filter();
```

Comment:
Did we test that filtering all frames is good?

Reply:
I have tested that on several examples. Result is the same.